pkg/lib/editor/editor.go

```diff
@@ -111,11 +121,49 @@ func RunConfigEditor(password, configPath, operatorEndpoint string, readOnlyFiel
 		fs.ServeHTTP(w, r)
 	})
 
+	// Create server base
 	s := &http.Server{
 		Addr:    ":" + strconv.Itoa(opts.port),
 		Handler: r,
 	}
 
-	log.Fatal(s.ListenAndServe())
+	// Try to load TLS
+	tlsConfig, err := loadTLS(opts.publicKeyPath, opts.privateKeyPath)
+	switch err == nil {
+	case true:
+		s.TLSConfig = tlsConfig
+		log.Printf("Running the configuration editor with HTTPS on port %v with username %s", opts.port, opts.username)
+		log.Fatal(s.ListenAndServeTLS("", ""))
+	default:
+		log.Printf("An error occurred loading TLS: " + err.Error() + ". Server falling back to HTTP.")
+		log.Printf("Running the configuration editor with HTTP on port %v with username %s", opts.port, opts.username)
+		log.Fatal(s.ListenAndServe())
+	}
+
+}
+
+// tlsConfig will attempt to create a tls config given a public and private key. It returns an error if it fails to create a Config.
+func loadTLS(publicKeyPath, privateKeyPath string) (*tls.Config, error) {
+
+	if publicKeyPath == "" {
+		return nil, errors.New("No public key provided for HTTPS")
+	}
+	if privateKeyPath == "" {
+		return nil, errors.New("No private key provided for HTTPS")
+	}
+	crt, err := ioutil.ReadFile(publicKeyPath)
+	if err != nil {
+		return nil, errors.New("Could not open public key: " + publicKeyPath)
+	}
+	key, err := ioutil.ReadFile(privateKeyPath)
+	if err != nil {
+		return nil, errors.New("Could not open private key: " + privateKeyPath)
+	}
+	cert, err := tls.X509KeyPair(crt, key)
+	if err != nil {
+		return nil, errors.New("Could not load X509 key pair: " + err.Error())
+	}
 
+	return &tls.Config{
+		Certificates: []tls.Certificate{cert}}, nil
```

Comment:
For your purposes, do you also need to add the system cert pool? https://golang.org/src/crypto/x509/cert_pool.go?s=1628:1668#L49 these are the trusted root certs located on the server.

Reply:
nope, not needed for this server. root certs are used at a later point but only when creating transports to external services.